weasel/schemas.py

```diff
@@ -78,6 +78,7 @@ class ProjectConfigSchema(BaseModel):
     commands: List[ProjectConfigCommand] = Field([], title="Project command shortucts")
     title: Optional[str] = Field(None, title="Project title")
     spacy_version: Optional[StrictStr] = Field(None, title="spaCy version range that the project is compatible with")
+    weasel_version: Optional[StrictStr] = Field(None, title="Weasel version range that the project is compatible with")
```

Comment:
We could handle the entire validation mechanism from here:

```python
    @validator("weasel_version")
    def validate_weasel_version(cls, version: Optional[StrictStr]) -> Optional[StrictStr]:

        if version and not is_compatible_version(about.__version__, version):
            err = (
                f"The {PROJECT_FILE} specifies a Weasel version range ({version}) "
                f"that's not compatible with the version of Weasel you're running "
                f"({about.__version__}). You can edit version requirement in the "
                f"{PROJECT_FILE} to load it, but the project may not run as expected."
            )
            msg.fail(err, exits=1)

        return version
```

Reply:
Right, maybe that's nice? Do you want to do that as part of this PR, or the next?

Reply:
I'd rather do a "Pydantic" PR if that's ok :)